### PR TITLE
DATA-2677 - Allow user to create dataset file from exported data for local testing of custom training scripts

### PR DESCRIPTION
**Pull request by tahiyasalam**
This new CLI commands allows a user to create a dataset file, where the file is formatted in the same way that's expected by the custom training script interface. This will allow a user to locally test parsing a dataset file and ensuring that their model compiles when writing training scripts. 

I manually tested a variety of scenarios and then used the dataset file as input to a training script. I tested with base command
`go run cli/viam/main.go dataset download --destination=test-download --dataset-id=6515b637a420eb685e988e84`
- with include-jsonl set to true, false, and unspecified which downloads the data as well the dataset, just the data, and just the data, respectively
- with parallel downloads set to 50, 100, and unspecified which works as implied

### CODE REVIEW

File: cli/dataset.go [hunk 1]
```diff
@@ -125,7 +131,7 @@ func (c *viamClient) listDatasetByOrg(orgID string) error {
 	return nil
 }
 
-// DatasetDeleteAction is the corresponding action for 'dataset rename'.
+// DatasetDeleteAction is the corresponding action for 'dataset delete'.
```
Comment (etai-shuchatowitz) [inline cli/dataset.go:134]: Nice catch!

File: cli/dataset.go [hunk 2]
```diff
@@ -150,3 +156,145 @@ func (c *viamClient) deleteDataset(datasetID string) error {
...
+func (c *viamClient) downloadDataset(dst, datasetID string, includeJSONLines bool, parallelDownloads uint) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+
+	var datasetFile *os.File
+	var err error
+	if includeJSONLines {
+		datasetPath := filepath.Join(dst, "dataset.jsonl")
+		if err := os.MkdirAll(filepath.Dir(datasetPath), 0o700); err != nil {
```
Comment (etai-shuchatowitz) [inline cli/dataset.go:183]: Probably a dumb question but why is this necessary? Isn't this gonna be just as necessary whether we need the jsonl or not?

File: cli/dataset.go [hunk 3]
```diff
@@ -150,3 +156,145 @@ func (c *viamClient) deleteDataset(datasetID string) error {
...
+	YMaxNormalized  float64 `json:"y_max_normalized"`
+}
+
+func binaryDataToJSONLines(ctx context.Context, client datapb.DataServiceClient, file *os.File,
+	id *datapb.BinaryID,
+) error {
+	var resp *datapb.BinaryDataByIDsResponse
+	var err error
+	for count := 0; count < maxRetryCount; count++ {
+		resp, err = client.BinaryDataByIDs(ctx, &datapb.BinaryDataByIDsRequest{
```
Comment (etai-shuchatowitz) [inline cli/dataset.go:243]: What's the reason we're using `ByIDs` but only passing in one ID?

**Review by etai-shuchatowitz — COMMENTED** [review]
Overall looks great! Tested locally with --jsonl true and false and got my expected results. Just left a few questions

File: cli/dataset.go (hunk 2 above)
Comment (tahiyasalam) [reply cli/dataset.go:183]: This just seems to be a paradigm used everywhere (e.g. [here](https://github.com/viamrobotics/rdk/blob/3fc61c36d9564dc85b5d908c3ebfaaa526c0a27c/cli/data.go#L462)). I'm also not super sure, but I wonder if the redundancy is to just make sure that the writer does not run into an perms error.

File: cli/dataset.go (hunk 3 above)
Comment (tahiyasalam) [reply cli/dataset.go:243]: Good q, this is mainly to use the batch operations from when we get data from the ByFilter API and then use the IDs to actually get the binary itself [here](https://github.com/viamrobotics/rdk/blob/3fc61c36d9564dc85b5d908c3ebfaaa526c0a27c/cli/dataset.go#L198). I think originally we went 1-1 for simplicity and to avoid timeouts associated with downloading many/larger files.

File: cli/dataset.go [hunk 4]
```diff
@@ -2,18 +2,24 @@ package cli
...
-	datasetFlagDatasetID  = "dataset-id"
-	datasetFlagDatasetIDs = "dataset-ids"
-	dataFlagLocationID    = "location-id"
-	dataFlagFileIDs       = "file-ids"
+	datasetFlagName             = "name"
+	datasetFlagDatasetID        = "dataset-id"
+	datasetFlagDatasetIDs       = "dataset-ids"
+	dataFlagLocationID          = "location-id"
+	dataFlagFileIDs             = "file-ids"
+	datasetFlagIncludeJSONLines = "jsonl"
```
[the remark this answers is not in the document]
Comment (tahiyasalam) [reply cli/dataset.go]: Updated!

File: cli/dataset.go (hunk 3 above)
Comment (etai-shuchatowitz) [reply cli/dataset.go:243]: Gotcha! Makes sense! Maybe just add a comment then

Comment (tahiyasalam) [reply cli/dataset.go:243]: Let me know if [this](https://github.com/viamrobotics/rdk/blob/3fc61c36d9564dc85b5d908c3ebfaaa526c0a27c/cli/data.go#L287) existing comment makes sense to you or you would like to see something else!

Comment (etai-shuchatowitz) [reply cli/dataset.go:243]: Ah, missed this! OK! I'll approve :)

**Review by etai-shuchatowitz — APPROVED** [review]
